src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueOfJsonPrimitive.cs

```diff
@@ -0,0 +1,241 @@
+                case JsonTokenType.String:
+                    byte[] quotedStringValue;
+
+                    if (reader.HasValueSequence)
+                    {
+                        // Throw if the long to int conversion fails. This can be accommodated in the future if needed by
+                        // adding another derived class that has a backing ReadOnlySequence<byte> instead of a byte[].
+                        int bufferLength = checked((int)(2 + reader.ValueSequence.Length));
+                        quotedStringValue = new byte[bufferLength];
+                        reader.ValueSequence.CopyTo(quotedStringValue.AsSpan().Slice(1, bufferLength - 2));
+                    }
+                    else
+                    {
+                        quotedStringValue = new byte[2 + reader.ValueSpan.Length];
+                        reader.ValueSpan.CopyTo(quotedStringValue.AsSpan(1, reader.ValueSpan.Length));
+                    }
+
+                    quotedStringValue[0] = quotedStringValue[quotedStringValue.Length - 1] = JsonConstants.Quote;
+                    return new JsonValueOfJsonString(quotedStringValue, reader.ValueIsEscaped, options);
+                case JsonTokenType.Number:
+                    byte[] numberValue = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan.ToArray();
+                    return new JsonValueOfJsonNumber(new JsonValueOfJsonNumber.JsonNumber(numberValue), options);
+                default:
+                    Debug.Fail("Only primitives allowed.");
+                    ThrowHelper.ThrowJsonException();
+                    return null!; // Unreachable, but required for compilation.
+            }
+        }
+
+        private sealed class JsonValueOfJsonString : JsonValue
+        {
+            private readonly byte[] _value;
+            private readonly bool _isEscaped;
+            private readonly JsonSerializerOptions _serializerOptions;
+
+            internal JsonValueOfJsonString(byte[] value, bool isEscaped, JsonSerializerOptions options)
+                : base(options.GetNodeOptions())
+            {
+                _value = value;
+                _isEscaped = isEscaped;
+                _serializerOptions = options;
+            }
+
+            private ReadOnlySpan<byte> ValueWithoutQuotes => _value.AsSpan(1, _value.Length - 2);
+
+            internal override JsonNode DeepCloneCore() => new JsonValueOfJsonString(_value, _isEscaped, _serializerOptions);
+            private protected override JsonValueKind GetValueKindCore() => JsonValueKind.String;
+
+            public override void WriteTo(Utf8JsonWriter writer, JsonSerializerOptions? options = null)
+            {
+                ArgumentNullException.ThrowIfNull(writer);
+
+                if (_isEscaped)
+                {
+                    writer.WriteStringValue(JsonReaderHelper.GetUnescapedString(ValueWithoutQuotes));
```

Comment:
Case in point, this representation is forcing a new allocation and unescaping cycle every time this value is being written out.